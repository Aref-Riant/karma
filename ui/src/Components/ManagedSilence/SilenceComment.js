import React from "react";
import PropTypes from "prop-types";

import Truncate from "react-truncate";

import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faExternalLinkAlt } from "@fortawesome/free-solid-svg-icons/faExternalLinkAlt";

import { APISilence } from "Models/API";

const SilenceComment = ({ silence, collapsed, afterUpdate }) => {
  const comment = (
    <Truncate className="font-italic" lines={collapsed ? 2 : false}>
      {silence.comment}
    </Truncate>
  );
  if (silence.jiraURL) {
    return (
      <a href={silence.jiraURL} target="_blank" rel="noopener noreferrer">
        <FontAwesomeIcon className="mr-2" icon={faExternalLinkAlt} />
        {comment}
      </a>
    );
  }
  return <React.Fragment>{comment}</React.Fragment>;
};
SilenceComment.propTypes = {
  silence: APISilence.isRequired,
  collapsed: PropTypes.bool.isRequired
};

export { SilenceComment };
