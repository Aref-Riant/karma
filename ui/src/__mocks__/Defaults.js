const DefaultsBase64 =
  "eyJSZWZyZXNoIjo0NTAwMDAwMDAwMCwiSGlkZUZpbHRlcnNXaGVuSWRsZSI6ZmFsc2UsIkNvbG9yVGl0bGViYXIiOmZhbHNlLCJEYXJrTW9kZSI6ZmFsc2UsIk1pbmltYWxHcm91cFdpZHRoIjo1NTUsIkFsZXJ0c1Blckdyb3VwIjoxNSwiQ29sbGFwc2VHcm91cHMiOiJleHBhbmRlZCJ9Cg==";
const DefaultsObject = {
  Refresh: 45000000000,
  HideFiltersWhenIdle: false,
  ColorTitlebar: false,
  DarkMode: false,
  MinimalGroupWidth: 555,
  AlertsPerGroup: 15,
  CollapseGroups: "expanded"
};

export { DefaultsBase64, DefaultsObject };
