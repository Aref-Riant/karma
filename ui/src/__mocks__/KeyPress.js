const PressKey = (tree, key, code) => {
  tree.simulate("keyDown", { key: key, keyCode: code, which: code });
  tree.simulate("keyUp", { key: key, keyCode: code, which: code });
};

export { PressKey };
