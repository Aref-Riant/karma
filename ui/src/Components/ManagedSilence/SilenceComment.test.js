import React from "react";

import { mount } from "enzyme";

import toDiffableHtml from "diffable-html";

import { MockSilence } from "__mocks__/Alerts";
import { AlertStore } from "Stores/AlertStore";
import { SilenceComment } from "./SilenceComment";

let silence;
let alertStore;

beforeEach(() => {
  silence = MockSilence();
  alertStore = new AlertStore([]);
});

afterEach(() => {
  jest.restoreAllMocks();
  fetch.resetMocks();
});

const CollapseMock = jest.fn();

const MountedSilenceComment = collapsed => {
  return mount(
    <SilenceComment
      alertStore={alertStore}
      alertCount={123}
      silence={silence}
      collapsed={collapsed}
      collapseToggle={CollapseMock}
    />
  );
};

describe("<SilenceComment />", () => {
  it("Matches snapshot when collapsed", () => {
    const tree = MountedSilenceComment(true);
    expect(toDiffableHtml(tree.html())).toMatchSnapshot();
  });

  it("Matches snapshot when expanded", () => {
    const tree = MountedSilenceComment(false);
    expect(toDiffableHtml(tree.html())).toMatchSnapshot();
  });

  it("Renders a JIRA link if present", () => {
    silence.jiraURL = "http://localhost/1234";
    silence.jiraID = "1234";
    const tree = MountedSilenceComment(true);
    expect(tree.find("a[href='http://localhost/1234']")).toHaveLength(1);
  });

  it("Renders a JIRA link if present and comment is expanded", () => {
    silence.jiraURL = "http://localhost/1234";
    silence.jiraID = "1234";
    const tree = MountedSilenceComment(false);
    expect(tree.find("a[href='http://localhost/1234']")).toHaveLength(1);
  });

  it("collapseToggle is called when collapse icon is clicked", () => {
    const tree = MountedSilenceComment(true);
    const collapse = tree.find("svg.fa-chevron-up");
    collapse.simulate("click");
    expect(CollapseMock).toHaveBeenCalled();
  });
});
