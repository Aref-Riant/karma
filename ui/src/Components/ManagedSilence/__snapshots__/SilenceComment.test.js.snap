// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SilenceComment /> Matches snapshot when collapsed 1`] = `
"
<span class=\\"font-italic\\"
      width=\\"0\\"
>
  <span>
  </span>
  <span>
    Mocked Silence
  </span>
  <span style=\\"position: fixed; visibility: hidden; top: 0px; left: 0px;\\">
    …
  </span>
</span>
"
`;

exports[`<SilenceComment /> Matches snapshot when expanded 1`] = `
"
<span class=\\"font-italic\\"
      width=\\"0\\"
>
  <span>
  </span>
  <span>
    Mocked Silence
  </span>
  <span style=\\"position: fixed; visibility: hidden; top: 0px; left: 0px;\\">
    …
  </span>
</span>
"
`;
